### PR TITLE
Block routes by role

### PR DESCRIPTION
**Pull request by pedrofp4444**
That's my first pull request. I tried to do something about the first card in TODO list. It's just a trial experience to get me familiarized with the platform.

### CODE REVIEW

**Comment by ruioliveira02** [comment]
You are a member of the organization, you don't need to fork the repository to push to it

File: lib/bokken_web/controllers/organizer_controller.ex
Comment (ruioliveira02) [inline lib/bokken_web/controllers/organizer_controller.ex]: I don't think it makes sense to block the index and show routes. Those should be accessible to everyone in the platform. What do you think @RuiL1904 ?

**Comment by pedrofp4444** [comment]
I wasn't a member when I did those changes.

File: lib/bokken_web/controllers/organizer_controller.ex
Comment (ruioliveira02) [inline lib/bokken_web/controllers/organizer_controller.ex]: We should probably create a file just for these guards, otherwise we are going to end up with a lot of duplicated code

File: lib/bokken_web/controllers/organizer_controller.ex
Comment (ruilopesm) [reply lib/bokken_web/controllers/organizer_controller.ex]: Yap, I agree. It would be a nice addition to list all the organizers on the dashboard, for example.

**Comment by pedrofp4444** [comment]
I'm a bit lost... How can I ask for review for the incomings changes?

**Review by ruilopesm — CHANGES_REQUESTED** [review]
Make sure the CI passes 🙏🏼

**Review by pedrofp4444 — COMMENTED** [review]
Now it's corrected.

File: lib/bokken/guards.ex
Comment (ruilopesm) [inline lib/bokken/guards.ex]:
```suggestion
  A module to handle guards used on the whole project.
```

File: lib/bokken/guards.ex
Comment (ruilopesm) [inline lib/bokken/guards.ex]:
```suggestion
  Defines the different types of Guards.
```

File: lib/bokken/guards.ex
Comment (ruilopesm) [inline lib/bokken/guards.ex]:
```suggestion
      Means to return True when a token has expired.
```

File: lib/bokken_web/controllers/file_controller.ex
Comment (ruilopesm) [inline lib/bokken_web/controllers/file_controller.ex]: Anyone should be able to create a file 💪🏼

File: lib/bokken_web/controllers/file_controller.ex
Comment (ruilopesm) [inline lib/bokken_web/controllers/file_controller.ex]: And to delete it.

File: lib/bokken_web/controllers/file_controller.ex
Comment (ruilopesm) [inline lib/bokken_web/controllers/file_controller.ex]: Same for the update.

File: lib/bokken_web/controllers/guardian_controller.ex
Comment (ruilopesm) [inline lib/bokken_web/controllers/guardian_controller.ex]: An organizer should also be able to create a guardian.

File: lib/bokken_web/controllers/guardian_controller.ex
Comment (ruilopesm) [inline lib/bokken_web/controllers/guardian_controller.ex]: And to update it too.

File: lib/bokken_web/controllers/lecture_controller.ex
Comment (ruilopesm) [inline lib/bokken_web/controllers/lecture_controller.ex:15]: I'm not sure about this. @ruioliveira02 Can I hear your opinion? 🙏🏼
Same for the other endpoints of this controller.

File: lib/bokken_web/controllers/pairing_controller.ex
Comment (ruilopesm) [inline lib/bokken_web/controllers/pairing_controller.ex]: This controller should only be able to an organizer. So that, put the `is_organizer/1` guard back 🙏🏼

File: lib/bokken_web/controllers/lecture_controller.ex
Comment (ruioliveira02) [reply lib/bokken_web/controllers/lecture_controller.ex:15]: With our current workflow, only organizers should be able to create lectures.

**Comment by ruioliveira02** [comment]
To add to what @RuiL1904 said, an organizer should have access to every single endpoint

**Comment by pedrofp4444** [comment]
Every review corrected and added organizer access capacity to every endpoint (create ninja params in this case).

**Review by ruilopesm — DISMISSED** [review]
Since this is an important feature, I would like to get a review from @ruioliveira02 too 🙏🏼

File: lib/bokken_web/controllers/lecture_controller.ex
Comment (ruioliveira02) [inline lib/bokken_web/controllers/lecture_controller.ex]: Mentors should be able to edit lectures

**Review by ruioliveira02 — CHANGES_REQUESTED** [review]
There is probably more changes to be made, we will need to test further to verify everything is working

File: lib/bokken_web/controllers/team_controller.ex
Comment (nelsonmestevao) [inline lib/bokken_web/controllers/team_controller.ex]: Maybe we could move this import to `lib/bokken_web.ex` to `controller`. What do you guys think?

File: lib/bokken_web/controllers/lecture_controller.ex
Comment (pedrofp4444) [reply lib/bokken_web/controllers/lecture_controller.ex]: Can they delete them?

File: lib/bokken_web/controllers/team_controller.ex
Comment (pedrofp4444) [reply lib/bokken_web/controllers/team_controller.ex]: Right, I'm going to implement it 👍

File: lib/bokken_web/controllers/lecture_controller.ex
Comment (pedrofp4444) [reply lib/bokken_web/controllers/lecture_controller.ex]: @ruioliveira02

**Review by ruioliveira02 — APPROVED** [review]
I think everything is working ok (at least what we are using). Before merging I would like another reviewer to approve @RuiL1904

File: lib/bokken_web/controllers/lecture_controller.ex
Comment (ruioliveira02) [reply lib/bokken_web/controllers/lecture_controller.ex]: No

**Review by ruilopesm — APPROVED** [review]
Congratz for your first contribution with us 🥳

**Pull request merged**
